src/app/cube/core/route-backwards-compat.guard.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot, UrlTree, Router } from '@angular/router';
+import { Observable } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+import { environment } from '@env/environment';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RouteBackwardsCompatGuard implements CanActivate {
+
+  constructor(private router: Router, private http: HttpClient) {}
+
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+      const cuidRegex = /([a-z0-9]){8}-([a-z0-9]){4}-([a-z0-9]){4}-([a-z0-9]){4}-([a-z0-9]){12}/;
```

Comment:
We can simply check if the string contains a hyphen instead of performing a complex regex comparison.

Reply:
Updated, I made it check to see if the first hyphen in the cuid is where it is supposed to be, i.e. index at 8.